galley/pkg/config/analysis/local/source.go

```diff
@@ -78,10 +79,13 @@ func (ph *precedenceHandler) Handle(e event.Event) {
 // For each collection, we want to only send this once, after all upstream sources have sent theirs.
 func (ph *precedenceHandler) handleFullSync(e event.Event) {
 	col := e.Source.Name()
+	ph.src.countMu.Lock()
 	ph.src.expectedCounts[col]--
 	if ph.src.expectedCounts[col] > 0 {
+		ph.src.countMu.Unlock()
```

Comment:
Nit: Code with two Unlocks takes brainpower to review, and have a strong preference for

```
ph.src.countMu.Lock()
ph.src.expectedCounts[col]--
shallHandle := ph.src.expectedCounts[col] > 0
ph.src.countMu.Unlock()
if (shallHandle) { ph.src.handler.Handle(e) }
```

(If we do it your way in the rest of Istio feel free to ignore this comment.)